main/lsp/requests/rename.cc

```diff
@@ -215,6 +215,43 @@ class ConstRenamer : public AbstractRenamer {
     }
 };
 
+class FieldRenamer : public AbstractRenamer {
+public:
+    FieldRenamer(const core::GlobalState &gs, const LSPConfiguration &config, const string oldName,
+                 const string newName)
+        : AbstractRenamer(gs, config, oldName, newName) {}
+    ~FieldRenamer() {}
+
+    void rename(unique_ptr<core::lsp::QueryResponse> &response, const core::SymbolRef originalSymbol) override {
+        auto loc = response->getLoc();
+        auto source = loc.source(gs);
+        if (!source.has_value()) {
+            return;
+        }
+
+        string newsrc;
+
+        // If the source includes `@` in the first character, then we're renaming the instance variable and need to make
+        // sure to include it. Otherwise, we're renaming an attr_reader/writer/accessor and need to make sure to not
+        // include it. This allows users to not care about whether they need to add the `@` themselves.
+        if (source.value()[0] == '@' && newName[0] != '@') {
```

Comment:
Can we exit early if `source.value().empty()`, or use `absl::StartsWith`?

There are a lot of places in Sorbet where we associate zero-width Loc's with pieces of syntax.